### PR TITLE
feat: add amazonbedrock

### LINKED ISSUE #716: [Feature]: AmazonBedrock AI provider
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

I use AWS EKS  and want use Amazon Bedrock Claude model, so I fork repo and write it. 

### Solution Description

Amazon Bedrock AI provider 

### Benefits

If you use AWS EKS, you can use k8sgpt from the Amazon Bedrock AI provider, and your data will only be kept in the Amazon VPC

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by stevensu1977**
Closes #716

## 📑 Description

Add AmazoneBedrock AI provider 
## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information
If you add amazonbedrock as ai provider , you should setup AWS AccessKey/SecretKey or IAM Role 

Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: pkg/ai/amzonbedrock.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	// Return the default model if the provided model is not in the list
+	return BEDROCK_MODELS[0]
+}
+
+// Configure configures the AmazonBedRockClient with the provided configuration and language.
+func (a *AmazonBedRockClient) Configure(config IAIConfig, language string) error {
+
+	// Create a new AWS session
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(BEDROCK_REGION),
```
Comment (AlexsJones) [inline pkg/ai/amzonbedrock.go:67]: We could modify the initialisation of the AI clients 
https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/ai/iai.go#L54
to support an argument passed through, what do you think?

**Review by AlexsJones — CHANGES_REQUESTED** [review]

**Comment by AlexsJones** [comment]
Awesome PR, please consider the parameterisation request of the region, or I can do it after @stevensu1977

File: pkg/ai/amzonbedrock.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		return "", nil
+	}
+
+	return response, nil
+}
+
+// GetName returns the name of the AmazonBedRockClient.
+func (a *AmazonBedRockClient) GetName() string {
+	return "amazonbedrock"
+}
```
Comment (psturc) [inline pkg/ai/amzonbedrock.go:162]: nit: typo `pkg/ai/amzonbedrock.go` -> `pkg/ai/amazonbedrock.go`

File: pkg/ai/amzonbedrock.go (hunk 1 above)
Comment (stevensu1977) [reply pkg/ai/amzonbedrock.go:67]: Sounds good , I try add it , may also can add modelID ,

Comment (stevensu1977) [reply pkg/ai/amzonbedrock.go]: @AlexsJones  , I already update source code,  add  -r, --providerRegion string   option to auth cmd  , but I not set it  as required filed ,  if not set it , we use  default region "us-east-1" region , if set invalid region also use default region .

....
providerRegion := GetRegionOrDefault(config.GetProviderRegion())
....

File: pkg/ai/amzonbedrock.go (hunk 2 above)
Comment (stevensu1977) [reply pkg/ai/amzonbedrock.go]: already fix it .

**Comment by AlexsJones** [comment]
Added some docs

**Pull request merged**